addons/stock/models/product.py

```diff
@@ -177,8 +178,8 @@ def _compute_quantities_dict(self, lot_id, owner_id, package_id, from_date=False
             # Calculate the moves that were done before now to calculate back in time (as most questions will be recent ones)
             domain_move_in_done = [('state', '=', 'done'), ('date', '>', to_date)] + domain_move_in_done
             domain_move_out_done = [('state', '=', 'done'), ('date', '>', to_date)] + domain_move_out_done
-            moves_in_res_past = {product.id: product_qty for product, product_qty in Move._read_group(domain_move_in_done, ['product_id'], ['product_qty:sum'])}
-            moves_out_res_past = {product.id: product_qty for product, product_qty in Move._read_group(domain_move_out_done, ['product_id'], ['product_qty:sum'])}
+            moves_in_res_past = {product.id: product_qty for product, product_qty in MoveLine._read_group(domain_move_in_done, ['product_id'], ['quantity:sum'])}
+            moves_out_res_past = {product.id: product_qty for product, product_qty in MoveLine._read_group(domain_move_out_done, ['product_id'], ['quantity:sum'])}
```

Comment:
```suggestion
            moves_in_res_past = {product.id: product_qty for product, product_qty in Move._read_group(domain_move_in_done, ['product_id'], ['quantity:sum'])}
            moves_out_res_past = {product.id: product_qty for product, product_qty in Move._read_group(domain_move_out_done, ['product_id'], ['quantity:sum'])}
```
Why not use the `quantity` field from moves directly?

Reply:
That should work in theory, but the location might not be the same on the move and the move line.
The move line should have a child location of the move, but there are no restriction on the usage of the child location; You can make a move line with 5 qty in WH/Stock and another 5 qty in WH/Stock/Scrap in the same stock move.

It's not the best workflow, but I've seen some users do weird stuff like that before.
Actually, this config might not even work properly, quants on hands include WH/Stock/Scrap. So we could consider using the move directly, as it should work in *most* cases.